### PR TITLE
Update/timezones query component

### PR DESCRIPTION
**Pull request by retrofox**
This PR adds the `query-timezones` component implemented according to most recently handling data of calypso. More information about this: https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer/wpcom-http

#### Run client tests

- data-layer timezones tests
```
> npm run test-client client/state/data-layer/wpcom/timezones/test/
```

- data query-timezones tests
```
> npm run test-client client/components/data/query-timezones/test/
```

#### Some tests using chrome devtools

You could dispatch a timezones request action straight using the console executing the right action type. it can be achieved using the global `actionTypes` var as well.

```
> dispatch( { type: actionTypes.TIMEZONES_REQUEST } );
```

After dispatching the action, you could check that the request has been triggered taking a look to the `network` tab:

And you could confirm that the action is being successful dispatched, paying attention how the set of actions are triggered one by one in the following order:

`WPCOM_HTTP_REQUEST`,
`TIMEZONES_REQUEST`,
`TIMEZONES_REQUEST_SUCCESS`,
`TIMEZONES_RECEIVE`,

### CODE REVIEW

File: client/state/data-layer/wpcom-http/actions.js [hunk 1]
```diff
@@ -35,7 +37,7 @@ export const http = ( {
 	body,
 	method,
 	path,
-	query: { ...query, apiVersion },
+	query: { ...query, apiVersion, apiNamespace },
```
Comment (gwwar) [inline client/state/data-layer/wpcom-http/actions.js]: Let's split this out into it's own PR first

File: client/state/data-layer/wpcom/timezones/test/index.js [hunk 2]
```diff
@@ -2,76 +2,90 @@
...
 
 /**
  * Internal dependencies
  */
-import { useSandbox } from 'test/helpers/use-sinon';
-import useNock from 'test/helpers/use-nock';
+import { http } from 'state/data-layer/wpcom-http/actions';
+
+// import { useSandbox } from 'test/helpers/use-sinon';
+// import useNock from 'test/helpers/use-nock';
```
Comment (gwwar) [inline client/state/data-layer/wpcom/timezones/test/index.js]: Let's remove these lines

**Comment by retrofox** [comment]
Let's close this one atm. We are going to improve how the request is built for the `http()` layer.
